pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -226,6 +270,12 @@ func getCandidates(shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1beta1
 		candidates = determineCandidatesOfSameProvider(seedList, shoot)
 	case strategy == config.SameRegion:
 		candidates = determineCandidatesWithSameRegionStrategy(seedList, shoot)
+	case strategy == config.BestRegion:
+		candidates = determineCandidatesWithSameRegionStrategy(seedList, shoot)
+		if len(candidates) > 0 {
+			return candidates, nil
+		}
+		fallthrough
 	case strategy == config.MinimalDistance:
 		candidates = determineCandidatesWithMinimalDistanceStrategy(seedList, shoot)
```

Comment:
If a scheduler is used the "BestRegion" strategy the function determineCandidatesWithSameRegionStrategy() is always executed twice (once in the switch and once after the fallthrough in the func [determineCandidatesWithMinimalDistanceStrategy](https://github.com/gardener/gardener/blob/a3d1c24b0c042c241cc0a4030846baa5ebe6d16d/pkg/scheduler/controller/shoot/scheduler_control.go#L366)).  In my opinion we should avoid e.g by adjusting the  determineCandidatesWithMinimalDistanceStrategy() to not call determineCandidatesWithSameRegionStrategy(). In the "BestSeed" we would then just call the determineCandidatesWithMinimalDistanceStrategy() directly.

```suggestion
              candidates = determineCandidatesWithSameRegionStrategy(seedList, shoot)
		if len(candidates) > 0 {
			return candidates, nil
		}
		candidates = determineCandidatesWithMinimalDistanceStrategy(seedList, shoot)
```

Reply:
I've removed BestRegion again according to your overall remark.